layrry-config/src/main/java/org/moditect/layrry/config/LayersConfigLoader.java

```diff
@@ -28,14 +38,75 @@ public static LayersConfig loadConfig(Path layersConfigFile) {
         for (LayersConfigParser parser : parsers) {
             if (parser.supports(layersConfigFile)) {
                 try (InputStream inputStream = layersConfigFile.toUri().toURL().openStream()) {
-                    return parser.parse(inputStream);
-                }
-                catch (IOException e) {
+                    return parser.parse(replacePlaceholders(inputStream));
+                } catch (IOException e) {
                     throw new IllegalArgumentException("Unexpected error parsing config file. " + layersConfigFile, e);
                 }
             }
         }
         throw new IllegalArgumentException("Unsupported config format. " + layersConfigFile);
     }
 
+    private static InputStream replacePlaceholders(InputStream inputStream) {
+        // detect OS
+        OsDetector detector = new OsDetector();
+
+        Properties props = new Properties();
+        props.putAll(System.getProperties());
+        props.putAll(detector.getProperties());
+        // special case for JavaFX OS classifier
+        String javafxClassifier = resolveJavaFxClassifier(detector.get(Detector.DETECTED_NAME));
+        if (null != javafxClassifier) {
+            props.put("javafx.os.classifier", javafxClassifier);
```

Comment:
Extract a constant?

Reply:
More importantly though, why is that special case needed?

Reply:
Because openjfx maps `osx` to `mac`. They used to map `windows` to `win` IIRC.

Reply:
Scratch that, they still map `windows` to `win`. Have to fix it.

Reply:
Still thinking about that; could we perhaps make it so that one generic {{os}} placeholder is resolved differently, depending on the context where it is used? I.e. to the Maven plug-ins values by default, but if we detect this happens in the context of a JavaFX dependency, then resolve it to the JavaFX classifiers? Question is how to detect and express this context of course. Would add a bit of complexity for us, but simplify things for users. WDYT?

Reply:
How would we know the context? For JavaFx {{os}} would typically be used for the classifier. But we don't know that unless we explicitly parse the GAV coordinates, in which case using a template engine is not required at all, we'll be back to parsing each expression in isolation.

Also, what if there's a JavaFx module and a different native module such as `sample-osx-1.2.3.jar` which is defined as `:com:acme:sample-{{os}}:1.2.3`; in this case the context is the artifact id.

I think keeping them separate is the way to go for now.

Reply:
Yes, exactly, the GAV coordinates would be that context, and there could be some piece of logic somewhere that depending on known GAV resolves the placeholder accordingly, or falls back to the default. But agreed that this is something we could leave for another time.